ngraph/core/include/ngraph/function.hpp

```diff
@@ -150,6 +169,7 @@ namespace ngraph
         topological_sort_t m_topological_sorter;
 
         ResultVector m_results;
+        SinkVector m_sinks;
```

Comment:
In all places where we re-create the ngraph function using the constructor, we can lose m_sinks in the new instance.
e.g. the typical way to create a new function based on the previous:
func1 = make_shared<ngraph::function>(some_results, some_parameters);
func1->set_sinks(some_sinks)
func2 = make_shared<ngraph::function>(func1->get_results(), func2->get_parameters());
____ we do not add sinks in function here_____ 

This could potentially be a problem when we replace control_dependencies with this API in ie_ir_parser.cpp (line 196)
Please look at https://github.com/openvinotoolkit/openvino/pull/1752/files , similar changes were made there, it may be useful.

Reply:
Potentially yes, but you call get_results, get_parameters, almost the same you will need to do with sinks:
call func2->add_sinks(func1->get_sinks()). The difference only in sending sinks to separate function, not constructor. 
And you always can make this error calling get_results instead of get_leafs, for example.

Reply:
Can we just use a copy ctor for that? Can we add a new ctor for `Function` to be able to forward sinks? Is it not changing an old API, it's the adding a new one.

Reply:
You mean contructor like Function(const Function & f) that will copy inside results, parameters and sinks, right? It will be good solution.

Reply:
Yes, but I see that copy ctor is deleted for some reason. @ilyachur and @GlebKazantaev do you know the reason? Probably because it is unusual to build another function on the same graph that was already used to define existing function. If we still think that this limitation is a valid one, then the example from @itikhono is not very realistic. And then, yes, user should maintain sinks manually. But anyway it is possible backward compatibility when you need to modify old code to correctly create a function that contains Assigns.